src/lib.rs

```diff
@@ -427,28 +433,45 @@ impl Check {
 #[non_exhaustive]
 #[derive(Debug)]
 pub struct CrateReport {
+    /// Bump between the current version and the baseline one
+    detected_bump: ActualSemverUpdate,
     /// Minimum bump required to respect semver.
+    /// For example, this is [`ReleaseType::Major`] if the crate contains
+    /// breaking changes.
     required_bump: ReleaseType,
 }
 
 impl CrateReport {
     /// Construct a new `CrateReport` with a successful outcome.
-    pub(crate) fn new_successful() -> Self {
+    pub(crate) fn new(detected_bump: ActualSemverUpdate) -> Self {
         Self {
+            detected_bump,
             required_bump: ReleaseType::Patch,
```

Comment:
You're probably still in the process of tweaking things, so this note is more for me: we should remember to change `required_bump` here. Just because the run was successful doesn't mean `required_bump: ReleaseType::Patch` is correct. Either having it as an `Option` type or having it match `ActualSemverUpdate` might be best. We don't know the exact minimum required bump -- if we detected a minor bump, we don't run any of the minor-requiring checks and don't know whether minor was overkill or the minimal appropriate one.

Reply:
> having it match ActualSemverUpdate might be best.

we can't do it, because there's no equivalent of `ActualSemverUpdate::NotChanged`.
I turned it into an Option 👍 

Btw, thanks for your early review. Really helpful.